payu/models/access.py

```diff
@@ -89,61 +100,68 @@ def setup(self):
-
-                    # With later versions this file exists in the prior restart
-                    # path, but this was not always the case, so check, and if
-                    # not there use prior output path
-                    if not os.path.exists(prior_cpl_fpath):
-                        print('payu: warning: {0} missing from prior restart '
-                              'path; checking prior output.'.format(cpl_fname),
-                              file=sys.stderr)
-                        if not os.path.isdir(model.prior_output_path):
-                            print('payu: error: No prior output path; '
-                                  'aborting run.')
-                            sys.exit(errno.ENOENT)
-
-                        prior_cpl_fpath = os.path.join(model.prior_output_path,
-                                                       cpl_fname)
+                    # Read the start date from the restart date namelist.
+                    start_date_fpath = os.path.join(
+                        model.prior_restart_path,
+                        model.start_date_nml_name
+                    )
 
                     try:
-                        prior_cpl_nml = f90nml.read(prior_cpl_fpath)
-                    except IOError as exc:
-                        if exc.errno == errno.ENOENT:
-                            print('payu: error: {0} does not exist; aborting.'
-                                  ''.format(prior_cpl_fpath), file=sys.stderr)
-                            sys.exit(exc.errno)
-                        else:
-                            raise
-
-                    cpl_nml_grp = prior_cpl_nml[cpl_group]
-
-                    # The total time in seconds since the beginning of
-                    # the experiment.
-                    total_runtime = int(cpl_nml_grp[runtime0_key] +
-                                        cpl_nml_grp['runtime'])
-                    run_start_date = cal.date_plus_seconds(init_date,
-                                                           total_runtime,
-                                                           caltype)
+                        start_date_nml = f90nml.read(start_date_fpath)[
+                            model.cpl_group]
+                    except FileNotFoundError:
+                        print(
+                            "Missing restart date file for model "
+                            f"{model.model_type}",
+                            file=sys.stderr
+                        )
+                        raise
+
+                    # Experiment initialisation date
+                    init_date = cal.int_to_date(
+                        start_date_nml["init_date"]
+                    )
+
+                    # Start date of new run
+                    run_start_date = cal.int_to_date(start_date_nml["inidate"])
+
+                    # run_start_date must be after initialisation date
```

Comment:
Did you test this? Its possible this would just work ?

Reply:
I tried running with 
```    
inidate = 1010101
init_date = 10101
```
and then during `payu setup` forced `runtime0=0`, to test whether it is sufficient to set just the `inidate`.

Some strange things end up happening, e.g. in `iceout085`:
```
(calendar)  idate =      1010101
(calendar)  idate =      2010101
(calendar)  idate =      2010101
```
the year jumped by 100 years, which also impacted the output file names:
```
iceh.0201-01.nc
```
Values like `runtime0` are [showing up](https://github.com/ACCESS-NRI/cice4/blob/e7549ebd2044690a432cc67c1317c81cb194b750/source/ice_restart.F90#L330) in the cice code, and so figured that a safe approach for the new calculations would be to ensure that cice is still provided with the same values of `inidate`, `init_date`, and `runtime0` as before.

Reply:
Thanks :)